src/clients/base.rs

```diff
@@ -314,7 +314,7 @@ where
     /// [Reference](https://developer.spotify.com/documentation/web-api/reference/#/operations/get-an-artists-albums)
     fn artist_albums<'a>(
         &'a self,
-        artist_id: &'a ArtistId,
+        artist_id: &'a ArtistId<'_>,
```

Comment:
`artist_id: &'a ArtistId<'_>,` should be `artist_id: ArtistId<'_>,`?  Because we don't need the explicit life notation `'a`.

Reply:
I'm going to reply here, but this applies to several other review comments too (every paginated endpoint, not those with `_manual`):

This is something that we discussed earlier on this PR, see [these comments](https://github.com/ramsayleung/rspotify/pull/305#discussion_r884136530) and [this one](https://github.com/ramsayleung/rspotify/pull/305#issuecomment-1142580377). Omitting the `&` reference works in the `sync` case, but not in the `async` one (because the closure passed to `paginate` would return a `Future` that contains a reference to the `id` that is owned by the closure).

As I wrote [here](https://github.com/ramsayleung/rspotify/pull/305#issuecomment-1179620123), I found a potential solution to this problem, which can be found in https://github.com/eladyn/rspotify/commit/b2bfaff0cf8b9586cd0bb56c21eacca683de63ef. That solution is a bit more complex however and might be better suited for a follow-up PR?